functionary/prompt.py

```diff
@@ -31,147 +187,543 @@ def from_message(cls, message: Dict) -> EndToken:
+                        responses.append(
+                            self.get_text_delta_response(None, True, finish_reason)
+                        )
+                    responses.append(
+                        self.get_function_delta_response(
+                            current_state, "", True, False, finish_reason
+                        )
+                    )
+                    return current_state, responses
+        else:
+            assert current_state["response_type"] is not None
+            if (
+                delta_text == self.from_token
+            ):  # skip until reach <content> to check type of response
+                current_state["current_text"] = ""
+                current_state["skip_until_reach"] = self.content_token
+                current_state["response_type"] = None
+                return current_state, None
+            else:
+                if current_state["response_type"] == "function":
+                    return current_state, self.get_function_delta_response(
+                        current_state, delta_text, False, False, finish_reason
+                    )
+                else:  # response_type=text
+                    return current_state, self.get_text_delta_response(
+                        delta_text, True, finish_reason
+                    )
 
-    Args:
-        message (Dict): dictionary in old format ({"to": xxx, "content": xxx})
 
-    Returns:
-        Dict: dictionary in openai format
-    """
-    if "to" not in message:
-        return message
-    return {
-        "role": message["role"],
-        "content": None,
-        "function_call": {"name": message["to"], "arguments": message["content"]},
-    }
+def get_random_tool_call_id():
+    return "call_" + "".join(
+        [random.choice(string.ascii_letters + string.digits) for _ in range(24)]
+    )
 
 
-def get_prompt_from_messages(messages: List[Dict], functions: Optional[List[Dict]] = []) -> str:
-    """return the final prompt that will be used.
-    Args:
-        messages (List[Dict]): list of messages where each message is in the format of OpenAI
-        functions (Optional[List[Dict]]): list of functions where each function is in the format of OpenAI
+def get_default_prompt_template() -> PromptTemplate:
+    """Return default prompt template to be used
 
     Returns:
-        str: the final prompt that will be used.
+        _type_: _description_
     """
-    messages_clone = messages.copy()  # To avoid modifying the original list
-
-    if functions is None:
-        functions = []
-
-    messages_clone.insert(0, {"role": "system", "content": generate_schema_from_functions(functions)})
-    messages_clone.insert(1, {"role": "system", "content": SYSTEM_MESSAGE})
-
-    full_text = ""
-    for message in messages_clone:
-        full_text += get_text_from_message(message)
-    return full_text.strip()
+    return PromptTemplateV2.get_prompt_template()
```

Comment:
Should we change this to one of the args in argparser in server_vllm.py/server.py?

Reply:
for inference, the template would be automatically chosen based on tokenizer so no need to add that option in server.py, server_vllm.py. However, we can **add the argument for training**